packages/web-components/src/utils/date-utils.ts

```diff
@@ -220,11 +245,19 @@ export function validate(
   const leapYear = checkLeapYear(year);
   const maxDay = daysForSelectedMonth(leapYear, month);
 
+  if (component.required && (!year || !month || (!monthYearOnly && !day))) {
+    component.invalidYear = (!year || year < minYear || year > maxYear);
+    component.invalidMonth = (!month || month < minMonths || month > maxMonths);
+    component.invalidDay = monthYearOnly ? false : (!day || day < minMonths || day > maxDay);
+    addError(component, 'date-error');
+    return;
+  }
```

Comment:
Moved this function to the beginning since it touches _all_ the invalid properties.